### PR TITLE
Add test for goto definition on a property pattern's member name.

### LINKED ISSUE #9072: Goto def on complex pattern member name doesn't work.
Given this code:

```c#
    public class Expression { }
    public class Constant : Expression
    {
        public readonly int Value;
        public Constant(int Value)
        {
            this.Value = Value;
        }
    }
    public class Plus : Expression
    {
        public readonly Expression Left, Right;
        public Plus(Expression Left, Expression Right)
        {
            this.Left = Left;
            this.Right = Right;
        }
    }
    public class X
    {
        public static void Main()
        {
            Expression expr = null;
            if (expr is Plus { $$Left is Plus })
```

Goto def on Left (in the last line) does not work.

### PR DESCRIPTION
**Pull request by gafter**
Closes #9072 

@dotnet/roslyn-ide @CyrusNajmabadi Please review.

### CODE REVIEW

File: src/EditorFeatures/Test2/GoToDefinition/GoToDefinitionTests.vb [hunk 1]
```diff
@@ -615,20 +615,76 @@ class C
             Await TestAsync(workspace)
         End Function
 
-        <WpfFact, Trait(Traits.Feature, Traits.Features.GoToDefinition)>
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Test2/GoToDefinition/GoToDefinitionTests.vb]: Looks like you deleted an existing test.  Did you mean to copy/paste instead?

**Comment by CyrusNajmabadi** [comment]
:+1: If you add the old test back :)

File: src/EditorFeatures/Test2/GoToDefinition/GoToDefinitionTests.vb (hunk 1 above)
Comment (gafter) [inline src/EditorFeatures/Test2/GoToDefinition/GoToDefinitionTests.vb]: oops. Will fix.

**Comment by DustinCampbell** [comment]
A little late, but :+1:

**Pull request merged**